Debug/cli/cli.c

```diff
@@ -348,6 +398,51 @@ inline void CLI_tabCompletion()
 	}
 }
 
+inline int wrap(int kX, int const kLowerBound, int const kUpperBound)
+{
+	int range_size = kUpperBound - kLowerBound + 1;
+
+	if (kX < kLowerBound)
+		kX += range_size * ((kLowerBound - kX) / range_size + 1);
+
+	return kLowerBound + (kX - kLowerBound) % range_size;
+}
+
+inline void cli_saveHistory(char *buff) {
+	if (buff==NULL) {
+		//clear the item
+		CLIHistoryBuffer[CLIHistoryTail][0] = '\0';
+		return;
+	}
+
+	// Copy the line to the history
+	int i;
+	for (i=0; i<CLILineBufferCurrent; i++)
+	{
+		CLIHistoryBuffer[CLIHistoryTail][i] = CLILineBuffer[i];
+	}
+}
+
+void cli_retreiveHistory(int index) {
+	char *histMatch = CLIHistoryBuffer[index];
```

Comment:
If a variable is in []'s put spaces around it. Optional exceptions is text like this ["stuff"] or numbers.